### ISSUE TITLE
Add associated value of UIEdgeInset to AvatarPosition enum

### ISSUE THREAD

**Issue by SD10** [MessageKit/MessageKit#206]
### Reasoning
The `LabelAlignment` enum already has this capability. I also think it fits nicely with #201.
It's important to note that this is a "nice to have" and is blocked before #201 is complete.

I'm willing to accept a PR but it requires a refactor of `MessagesCollectionViewFlowLayout` and I'm not sure if the code is readable by anyone but me 😅

This will definitely add some complexity and I'll re-evaluate based on the amount of complexity that #201 adds.

**Comment by SD10** [comment]
Eh, this becomes a lot less complex after #580 but I'm still torn on adding it. On one hand, it makes sense because everything else is already supported.

Would basically allow users to modify the offset of the `AvatarView` in the same way as the `MessageContainerView` with `messagePadding`.

Only considering this now because I'm auditing every issue for 1.0

cc @zhongwuzw

**Comment by SD10** [comment]
Need to be able to set the padding on the side of the avatar opposite of the `MessageContainerView`

**Comment by chc24** [comment]
Hey @SD10! Any solution currently for wanting to add some inset to either side's avatar view?

**Comment by SD10** [comment]
We can't use `UIEdgeInset` here, but we can use the new `HorizontalEdgeInset` class. The reason is the top and bottom insets would affect the Y position of the `AvatarView`.

**Comment by nathantannar4** [comment]
@SD10 Is this still relevant?

**Comment by SD10** [comment]
Yeah it is @nathantannar4
